src/users.ts

```diff
@@ -0,0 +1,874 @@
+import Errors from './errors'
+
+export interface User {
+  name?: string
+  id?: string
+  credentials?: {
+    identity: string
+    secret?: string
+  }
+  owner?: string
+  tags?: [string, string]
+  role?: string
+  status?: 'enabled' | 'disabled'
+  createdAt?: string
+  updatedAt?: string
+  updatedBy?: string
+}
+
+interface PageRes {
+  total: number
+  offset: number
+  limit: number
+}
+
+interface UsersInterface {
+  users: User[]
+  page: PageRes
+}
+
+interface Groups {
+  groups: Groups[]
+  page: PageRes
+}
+
+interface Things {
+  things: Things[]
+  page: PageRes
+}
+
+interface Channels {
+  channel: Channels[]
+  page: PageRes
+}
+
+interface Login {
+  identity?: string
+  secret?: string
+  domain_id?: string
+}
+
+interface QueryParams {
+  offset: number
+  limit: number
+}
+
+interface Token {
+  access_token: string
+  refreshToken: string
+}
+
+interface Status {
+  status: string
+}
+
+export default class Users {
+  // Users API client
+  /**
+   * @class Users -
+   * Users API is used for creating and managing users.
+   * It is used for creating new users, logging in, refreshing tokens,
+   * getting user information, updating user information, disabling
+   * and enabling users.
+   * @param {String} usersUrl - URL to the Users service.
+   * @returns {Object} - Users object.
+   */
+  private readonly usersUrl: URL
+  private readonly contentType: string
+  private readonly usersEndpoint: string
+  private readonly userError: Errors
+
+  public constructor (usersUrl: string) {
+    this.usersUrl = new URL(usersUrl)
+    this.contentType = 'application/json'
+    this.usersEndpoint = 'users'
+    this.userError = new Errors()
+  }
+
+  public async Create (user: User, token?: string): Promise<User> {
+    // Creates a new user
+    /**
+     * @method Create - Creates a new user.
+     * @param {Object} user - User object.
+     * @param {String} token - Access token.
+     * @returns {Object} - User object.
+     * @example
+     * const user = {
+     * "credentials": {
+     *    "identity": "admin@example.com",
+     *   "password": "12345678"
+     * }
+     * }
+     */
+
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      },
+      body: JSON.stringify(user)
+    }
+
+    try {
+      const response = await fetch(
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+        new URL(this.usersEndpoint, this.usersUrl).toString(),
+        options
+      )
+      if (!response.ok) {
+        const errorRes = await response.json()
+        throw this.userError.HandleError(errorRes.error, response.status)
+      }
+      const userData = await response.json()
```

Comment:
Enforce type for these responses. for example this should be 
```suggestion
      const userData : User = await response.json()
```

do sth similar for the rest